### PR TITLE
fix: clean upp edge cases around simulator kwargs

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Fix input handling for device.run

*Testing done:*

unit + integ tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/aws/aws_quantum_task.py [hunk 1]
```diff
@@ -462,7 +462,7 @@ def _(
     *args,
     **kwargs,
 ) -> AwsQuantumTask:
-    if open_qasm_program.inputs is not None:
+    if open_qasm_program.inputs:
```
Comment (ajberdy) [inline src/braket/aws/aws_quantum_task.py:465]: This change is because the to_openqasm method now adds an empty dict instead of None

**Review by math411 — APPROVED** [review]
